### ISSUE TITLE
Synchronizing a Hornet 0.4.0/dev with an IRI 1.8.6-RC5 breaks solidification

### ISSUE THREAD

**Issue by luca-moser** [iotaledger/hornet#449]
**Describe the bug**
Letting an IRI 1.8.6-RC5 synchronize with a HORNET node leads to a massive amount of outgoing TPS (20-70k) to the IRI node and causes the HORNET node to no longer be able to sync up with its own peers: it gets stuck at the same milestone.

**To Reproduce**
- Let an IRI 1.8.6-RC5 sync up from a 0.4.0 HORNET node
- See the massive amount of outgoing TPS (probably just caused by IRI requesting the same things over and over again)
- See how the HORNET node no longer acknowledges new milestones and seems to be stuck.

**Expected behavior**
HORNET operates further without getting stuck.

**Environment information:**
 - Ubuntu 18.04
 - RAM: 32GB
 - Cores: 8
 - Type: CX51
 - HORNET version: 
```
ommit 7511e71774ccbf883a63f6053d1b9cf8076e8103 (HEAD -> boltdb/array-freelist, origin/develop, develop)
Author: Alexander Sporn <github@alexsporn.de>
Date:   Thu May 21 19:37:34 2020 +0200
```

**Additional context**
Besides that, the IRI node seems not to really synchronize but it is not clear whether IRI is just not requesting the right things or HORNET is only answering with the transaction IRI needs. Obviously the HORNET node had its pruning index lower than the milestone from which the IRI node started  from.

**Comment by luca-moser** [comment]
There seems to be a very large amount of replies trying to get locks in `replyToAllRequests()`: https://ybin.me/p/5e3a04d89ce31d44#IbpO4pgEjF3wEaXKFUrmGGLmi1KLf6wXJBb9y2hEuOI=

Note that when you disconnect the IRI node, HORNET seems to not be able to recognize further milestones for a couple of minutes and then suddenly is again capable of doing so.

**Comment by luca-moser** [comment]
Using an IRI 1.8.5 has the same effect only that instead of HORNET throwing out 20-70k TPS it is more in the realm of 10k steadily.

**Comment by muXxer** [comment]
Fixed with #453
